### PR TITLE
DR-13: Env-specific config, vault, local minikube setup with remote debug

### PR DESCRIPTION
**Pull request by thathert**
This started out being a branch about auth and then I realized I didn't have the right local setup to actually be able to test I wanted. The idea:

- it'd be nice to see what comes out of "the other side" of the proxy, so that we're not duplicating work
- instead of wiring things together with docker compose, we could get a better idea of how things would work in our projects by running kubernetes* locally
- if we are running things locally, it would be useful to be able to remote debug the API running inside of the local cluster
- if we run things in a local kubernetes cluster, we can probably reuse the same deployment code for our remote kubernetes clusters

(*) the closest thing to a local k8s cluster is minikube

So the result is:

- vault secrets for local and dev along with a ctmpl file for populating a kubernetes secrets file with base64 encoded secret values
- minikube running locally
- a special docker file that configures the Spring boot API for remote debugging
- pod specs for the oidc proxy, api, and a standalone postgres db
- service specs to wire everything together

There are still some rough edges. I'm not the hugest fan of how minikube handles "load balancers" or tunneling cluster IPs to work on the host IP, but this was a useful start. I think we can leave a lot of the spring security complexity out by locking things down with the OIDC proxy for now. We will have a header with the users bearer token and their email once they make it through the proxy, and that should be enough for us to start using SAM and getting the right audit logs.

### CODE REVIEW

File: ops/local/run.sh [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+rm -r $SCRATCH
+
+# finally, set up a tunnel to allow remote debugging and output the new entry for /etc/hosts
+API_CLUSTER_IP=$(kubectl get service api-service -o jsonpath='{.spec.clusterIP}')
+PROXY_CLUSTER_IP=$(kubectl get service oidc-proxy-service -o jsonpath='{.spec.clusterIP}')
+echo "remote debug available at ${API_CLUSTER_IP}:5005"
+echo "to use the oidc proxy, add this to your /etc/hosts file:"
+echo "${PROXY_CLUSTER_IP}   local.broadinstitute.org"
+echo "about to call minikube tunnel, it will ask you for your password and then fill up your console with logs.."
```
Comment (ddietterich) [inline ops/local/run.sh]: which password?

**Review by ddietterich — APPROVED** [review]
I'm not a competent reviewer for this. As far as I can tell it is fine.
Two thoughts:
- what are the consequences if there is a bug in this? I think it wouldn't matter much at this point because we could continue developing as we are doing now.
- at least one other person should learn this stuff in detail so we have at least two sets of competent eyes reviewing stuff in this space.

**Comment by thathert** [comment]
> I'm not a competent reviewer for this. As far as I can tell it is fine.
> Two thoughts:
> 
> * what are the consequences if there is a bug in this? I think it wouldn't matter much at this point because we could continue developing as we are doing now.
> * at least one other person should learn this stuff in detail so we have at least two sets of competent eyes reviewing stuff in this space.

If there is a bug, then it prevents you from running the API locally in minikube. You can still keep using your local setup (bootRun and/or docker).

@ahaessly has learned this from us pairing on getting it working this week. Happy to go over it in more detail if you or anyone else would like.

**Review by ddietterich — APPROVED** [review]
LGTM

**Review by ahaessly — APPROVED** [review]
still need a delay after starting postgres before psql can be used

**Pull request merged**